### PR TITLE
New funcionality and customiztaions for LDAPc

### PR DESCRIPTION
**Pull request by stavamichal**
- add new attribute bonaFideStatus to LDAP (add support for this
   attribute to LDAPc and LDAPc-Initializer)
 - use constants instead of texts where it is possible (prevent
   duplicity and typo errors in code)
 - add support of moving groups in Perun Vo structure (as subtrees)
 - minor cosmetic changes in code

### CODE REVIEW

**Comment by zlamalp** [comment]
Merge (deploy) waits for #1663, #1661

File: perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/EventProcessorImpl.java [hunk 1]
```diff
@@ -94,18 +95,83 @@
...
+	public static final String ldapAttrMail = perunAttrMail;
+	public static final String ldapAttrOrganization = "o";
+	public static final String ldapAttrTelephoneNumber = "telephoneNumber";
+	public static final String ldapAttrUserCertDNs = "userCertificateSubject";
+	public static final String ldapAttrBonaFideStatus = "bonaFideStatus";
+	public static final String ldapAttrSchacHomeOrganizations = perunAttrSchacHomeOrganizations;
+	public static final String ldapAttrEduPersonScopedAffiliations = perunAttrEduPersonScopedAffiliations;
+	public static final String ldapAttrLibraryIDs = perunAttrLibraryIDs;
+	public static final String ldapAttrUidNumber = "uidNumber;x-ns-";
+	public static final String ldapAttrLogin = "login;x-ns";
```
Comment (Vojtech-Sassmann) [inline perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/EventProcessorImpl.java]: This should probably be "login;x-ns-"

**Comment by stavamichal** [comment]
Manually tested on devel. It seems to be ok.

File: perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/EventProcessorImpl.java (hunk 1 above)
Comment (stavamichal) [reply perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/EventProcessorImpl.java]: Nice one, yes this definitely should be "ns-". Thank you. It is fixed now.

File: perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/EventProcessorImpl.java [hunk 2]
```diff
@@ -94,18 +95,83 @@
...
+	public static final String ldapAttrMemberOfPerunVo = "memberOfPerunVo";
+	public static final String ldapAttrEntryStatus = "entryStatus";
+	public static final String ldapAttrIsServiceUser = "isServiceUser";
+	public static final String ldapAttrIsSponsoredUser = "isSponsoredUser";
+
+	//LDAP OBJECT CLASSES
+	public static final String objectClassTop = "top";
+	public static final String objectClassPerunResource = "perunResource";
+	public static final String objectClassPerunGroup = "perunGroup";
+	public static final String objectClassoOrganization = "organization";
```
[the remark this answers is not in the document]
Comment (stavamichal) [reply perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/EventProcessorImpl.java]: Yes, it is typo. I fixed that. Thank you.

File: perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/EventProcessorImpl.java [hunk 3]
```diff
@@ -94,18 +95,83 @@
...
+	public static final String ldapAttrTelephoneNumber = "telephoneNumber";
+	public static final String ldapAttrUserCertDNs = "userCertificateSubject";
+	public static final String ldapAttrBonaFideStatus = "bonaFideStatus";
+	public static final String ldapAttrSchacHomeOrganizations = perunAttrSchacHomeOrganizations;
+	public static final String ldapAttrEduPersonScopedAffiliations = perunAttrEduPersonScopedAffiliations;
+	public static final String ldapAttrLibraryIDs = perunAttrLibraryIDs;
+	public static final String ldapAttrUidNumber = "uidNumber;x-ns-";
+	public static final String ldapAttrLogin = "login;x-ns-";
+	public static final String ldapAttrUserPassword = "userPassword";
+	public static final String ldapAttrSureName = "sn";
```
Comment (zlamalp) [inline perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/EventProcessorImpl.java]: Just fix all occurrence of surname to surname.

**Pull request merged**